src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -67,18 +67,6 @@
 // Debug level for DWC2
 #define DWC2_DEBUG    2
 
-#ifndef dcache_clean
-#define dcache_clean(_addr, _size)
```

Comment:
dcache code should be removed and add MPU suggestion to make usb ram non cacheable:
- Not all internal buffers are aligned to cache line size and has size of multiple cache line size, speculative read or cache eviction of adjacent variables could lead to racing with DMA.
- Frequent cache clean invalidate hurt performance
- In most cases Cortex-M7 recommend use DTCM where cache is not used

Reply:
At least for stm32f7xx, those all seem to have the AHB-S port of the M7 connected directly to the bus matrix, so it is accessible by the USB DMA in this case. Though, that is only for DTCM, ITCM is still inaccessible by anything besides the CPU.

Reply:
right, I haven't check f7, maybe this is specific to h7 only